.github/workflows/ci.yml

```diff
@@ -7,18 +7,72 @@ on:
 env:
   # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
   BUILD_TYPE: Release
-  GTEST: /home/runner/work/ssc/ssc/googletest
-  SSCDIR: /home/runner/work/ssc/ssc/ssc
 
 jobs:
-  build:
-    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
-    # You can convert this to a matrix build if you need cross-platform coverage.
-    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+  build-on-ubuntu1804:
     runs-on: ubuntu-18.04
 
     steps:
     
+    - name: Set relative paths
+      run: | 
+        GTEST=$GITHUB_WORKSPACE/googletest
+        echo "GTEST=$GTEST" >> $GITHUB_ENV
+        SSCDIR=$GITHUB_WORKSPACE/ssc
+        echo "SSCDIR=$SSCDIR" >> $GITHUB_ENV
+    
+    - name: Clone Gtest
+      uses: actions/checkout@v2
+      with:
+        repository: google/googletest
+        path: googletest
+    
+    - name: build Gtest
+      run: | 
+          export 
+          mkdir ${GTEST}/build
+          cd ${GTEST}/build
+          cmake -DCMAKE_CXX_FLAGS=-std=c++11 ..
+          make          
+    
+    - name: Checkout SSC
+      uses: actions/checkout@v2
+      with:
+         path: ssc
+
+    - name: Configure CMake
+      # Configure cmake to build ssc tests but not tools
+      run: |
+        mkdir ${SSCDIR}/build
+        cd ${SSCDIR}/build
+        cmake .. -DCMAKE_BUILD_TYPE=Release -DSAM_SKIP_TOOLS=1
+
+    - name: Build
+      # Build your program with the given configuration
+      run: |
+        cd ${SSCDIR}/build
+        make -j4
+
+    - name: Test
+      # Turn off fast fail for when the landbosse tests write to cerr
+      shell: bash
+      run: |
+        set -e
+        ${SSCDIR}/build/test/Test
+
+      
+  build-on-mac:
+    runs-on: macos-latest
```

Comment:
Do you know the mac runner hardware and macOS version? what is macos-latest?

Reply:
`macOS Catalina 10.15`, as per https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

Reply:
Apparently they're in the process of migrating: https://github.com/actions/virtual-environments/issues/4060
It'd probably safer to fix a version here, let me know which is preferred.

Reply:
For reference, the 2021.12.02 release was built using 11.6.1 on an Intel based Macbook Pro.
The macOS_wx3.1.5 was built using 12.0.1 on an Apple Silicon M1 based Macbook Air for Intel and Apple Silicon (using Rosetta) based Macs.